### PR TITLE
DDP-1307: Added backend route that provides list of languages support…

### PR DESCRIPTION
**Pull request by charlottedye**
…ed by a study and information about which is the default

## Context

DDP-1307 is the ticket for this pull request.  Pepper needs a route that will tell the frontend what languages have been configured for a particular study.  For example, Prion will support four languages and needs to show the different languages in a language selector component in the UI, but this functionality will be useful for multiple studies.  It is already possible to configure the available languages and specify which is the default, so this just makes it possible for the frontend to get that information.  Here is an example of possible output: [{"languageCode":"en","displayName":"English","isDefault":true,"studyId":1,"languageId":1}].  However, there could be any number of languages (or none at all).  I may need some help on getting this ready to actually merge in, but I did write automated tests and did test this manually to make sure it worked.  

## Checklist

- [x] I have labeled the type of changes involved using the `C-*` labels.
- [ ] I have assessed potential risks and labeled using the `R-*` labels.
- [x] I have considered error handling and alerts, and added `L-*` labels as needed.
- [x] I have considered security and privacy, and added `I-*` labels as needed
- [ ] I have analyzed my changes for stability, fault tolerance, graceful degradation, performance bottlenecks and written a brief summary in this PR.
- [ ] If applicable, I have discussed the analytics needs at both a platform and study level with Product and instrumented code accordingly.
-  [x] n/a If applicable, my UI/UX changes have passed muster with Product/Design via an over-the-shoulder review, screenshots, etc.

_If unsure or need help with any of the above items, add the `help wanted` label. For items that starts with `If applicable`, if it is not applicable, check it off and add `n/a` in front._

## FUD Score

_Overall, how are you feeling about these changes?_

- [ ] :relaxed: All good, business as usual!
- [x] :sweat_smile: There might be some issues here
- [ ] :scream: I'm sweaty and nervous

## How do we demo these changes?

_How does one observe these changes in a deployed system? Note that **user visible** encompasses many personas--not just patients and study staff, but also ops duty, your fellow devs, compliance, etc._

- [ ] They are user-visible in dev as a regular user journey and require no additional instructions.
- [x] Getting dev into a state where this is user-visible requires some tech fiddling. I have documented these steps in the related ticket.
- [ ] Requires other features before it's human visible. I have documented the blocking issues in jira.
-   [x] Not sure if I should be deploying this to dev first. I have no idea how to demo this. Please help me!  

## Testing

- [x] I have written automated positive tests
- [x] I have written automated negative tests
- [ ] I have written zero automated tests but have poked around locally to verify proper functionality
- [ ] The jira ticket has acceptance criteria and QA has the needed information to test changes

## Release

- [x] These changes require no special release procedures--just code!
- [ ] Releasing these changes requires special handling and I have documented the special procedures in the release plan document

### CODE REVIEW

File: pepper-apis/src/main/java/org/broadinstitute/ddp/db/dao/StudyLanguageDao.java
Comment (yufengwng) [inline pepper-apis/src/main/java/org/broadinstitute/ddp/db/dao/StudyLanguageDao.java]: Typically, caller should almost always check the returned value to ensure that indeed only one row is deleted. For convenience I recommend doing that check in this method itself so we lessen the burden on callers. There's the `DBUtils.checkDelete()` method you can use.

We don't have to do it for the method in `StudyLanguageSql` since it's more of a "lower-level" SQL method. But for DAOs, let's add a check.

File: pepper-apis/src/main/java/org/broadinstitute/ddp/db/dao/StudyLanguageDao.java
Comment (yufengwng) [inline pepper-apis/src/main/java/org/broadinstitute/ddp/db/dao/StudyLanguageDao.java]: Can we name it `findStudyLanguages()`? Or even just `findLanguages()`.

File: pepper-apis/src/main/java/org/broadinstitute/ddp/db/dao/StudyLanguageSql.java
Comment (yufengwng) [inline pepper-apis/src/main/java/org/broadinstitute/ddp/db/dao/StudyLanguageSql.java]: I don't think we need the single quotes. Typically, I prefer to leave the column names in snake-case and do the mapping in the Java class constructor itself using annotations.

File: pepper-apis/src/main/java/org/broadinstitute/ddp/db/dao/StudyLanguageSql.java
Comment (yufengwng) [inline pepper-apis/src/main/java/org/broadinstitute/ddp/db/dao/StudyLanguageSql.java]: Can we just `join` syntax here? E.g. `from study_language as sl join language_code as lc on ...`.

File: pepper-apis/src/main/java/org/broadinstitute/ddp/model/study/StudyLanguage.java
Comment (yufengwng) [inline pepper-apis/src/main/java/org/broadinstitute/ddp/model/study/StudyLanguage.java]: I recommend adding annotations to help facilitate JDBI mapping the SQL data properly:
```java
@JdbiConstructor
public StudyLanguage(
    @ColumnName("language_code") String languageCode,
    @ColumnName("name") String displayName,
    ...) {
}
```

File: pepper-apis/src/main/java/org/broadinstitute/ddp/model/study/StudyLanguage.java
Comment (yufengwng) [inline pepper-apis/src/main/java/org/broadinstitute/ddp/model/study/StudyLanguage.java:9]: Since this object is returned directly from the route, it means this object will get serialized to JSON. In that case, I tend to prefer adding `@SerializedName()` annotations to these fields to explicitly spell out the JSON property name:
```java
@SerializedName("languageCode")
private String languageCode
```

File: pepper-apis/src/main/java/org/broadinstitute/ddp/model/study/StudyLanguage.java
Comment (yufengwng) [inline pepper-apis/src/main/java/org/broadinstitute/ddp/model/study/StudyLanguage.java]: We typically don't return these database PK IDs in our JSON responses or expose in anyway via our APIs. Can you mark them as `transient` so they don't show up in JSON?
```java
private transient long studyId;
private transient long languageId;
```

File: pepper-apis/src/main/java/org/broadinstitute/ddp/route/ListStudyLanguagesRoute.java
Comment (yufengwng) [inline pepper-apis/src/main/java/org/broadinstitute/ddp/route/ListStudyLanguagesRoute.java]: Nice! Rule-of-thumb is for every route to start with some logging for the request. Can you turn this into `LOG.info()`?

**Review by yufengwng — COMMENTED** [review]
Looking good! Just a few recommendations.

**Comment by charlottedye** [comment]
Thanks for the review!  I'll make these changes and retest.

**Review by yufengwng — APPROVED** [review]
LGTM

**Comment by yufengwng** [comment]
Currently, PRs need to pass our tests via CircleCI. We don't have automation for that part so you'll need to manually trigger it from your local:
```
$ cd ddp-study-server/pepper-apis
$ ./scripts/run_ci_tests.sh <your_branch_name>
```

**Comment by yufengwng** [comment]
And your branch needs to be up-to-date with `develop` before merging. Sometimes clicking on the `Update branch` button on github or doing a manual merge/push will dismiss the approval checkmark you need (typically if there's merge conflicts). If that happens, ping me (or another PR reviewer) and I'll approve again.

**Comment by charlottedye** [comment]
It's saying I need a Circle CI token but I don't think I even have a Circle CI account

**Comment by yufengwng** [comment]
It should be your github account. Did you follow the link the little output from the script?

**Comment by charlottedye** [comment]
I did and it gave me an error about being logged out.  I'll connect it to my GitHub.

**Pull request merged**
